spec/services/participants/change_schedule/ecf_spec.rb

```diff
@@ -24,6 +24,25 @@ def matches_lead_provider?
   end
 
   describe "validations" do
+    context "when participant is withdrawn" do
+      let(:user) { profile.user }
+      let(:profile) { create(:mentor_participant_profile, status: :withdrawn) }
+      let(:schedule) { create(:ecf_mentor_schedule) }
+
+      subject do
+        Participants::ChangeSchedule::Mentor.new(params: {
+          schedule_identifier: schedule.schedule_identifier,
+          participant_id: user.id,
+          course_identifier: "ecf-induction",
+          cpd_lead_provider: CpdLeadProvider.new,
+        })
+      end
+
+      it "should have an error" do
+        expect { subject.call }.to raise_error(ActionController::ParameterMissing, /must be a valid Participant ID/)
+      end
```

Comment:
I would question whether the error message could be a bit more descriptive but I think we have a ticket for that

Reply:
yes the way the validations are order dependent is making this nearly impossible to tell the provider that this participant is withdrawn, which really you be the error.